paper.tex

```diff
@@ -71,7 +71,11 @@ \section{Introduction}
 
 Our central approach bounding the error tolerance is \emph{error metrics,} which predict the Fréchet distance error. The error metrics presented in this paper are both straightforward to compute, and also \emph{invertible}, which avoids excessive subdivision and also increases the amount of available parallelism to exploit. Previous approaches to error measurement generally follow a ``cut then measure'' approach, where the approximate curve is generated, then error measured by sampling. Such sampling is often time consuming, involving iteration, and even then risks underestimating the error due to inadequate sampling.
 
-This paper presents a comprehensive solution to stroke expansion, well suited to GPU implementation. It can produce both flattened polylines or an approximation consisting of circular arcs. The best choice depends on the capabilities of the path rendering mechanism following stroke expansion, but as we show, outlines consisting of circular arcs have many fewer segments and are correspondingly faster to produce. The algorithm is based on Euler spiral segments as an intermediate representation, with an iterative algorithm based on a straightforward error metric for conversion from cubic Béziers. We have also devised a compact binary encoding of paths, suitable for fully parallel computation of stroke outlines, while requiring minimal CPU-side processing. Our algorithm has been implemented in GPU compute shaders, integrated in a full rendering engine for vector graphics, and shows a dramatic speedup over CPU methods.
+The style of stroke rendering also optionally includes \emph{dashing.} Previous work, particularly \citet{Kilgard2020}, has explained how to break the input path into dash segments of the specified length on GPU using a prefix sum of segment arc lengths. There are implementation tradeoffs to GPU-based dash processing, including potentially a larger number of dispatch stages, and we were able to achieve acceptable performance with the dash processing on CPU. Thus, whether moving dashing to a GPU stage is a performance win depends on the scene and other factors in the overall rendering pipeline. Even so, for scenes that have a very large number of dashed strokes, we would expect further performance improvement from GPU-based dashing.
+
+Previous work, particularly \citet{Kilgard2020}, has explained how to process dashing patterns on GPU using a prefix sum of segment arc lengths. There are implementation tradeoffs to GPU-based dash processing, including potentially a larger number of dispatch stages, and we were able to achieve acceptable performance with the dash processing on CPU. Thus, whether moving dashing to a GPU stage is a performance win depends on the scene and other factors in the overall rendering pipeline. Even so, for scenes that have a very large number of dashed strokes, we would expect further performance improvement from GPU-based dashing.
```

Comment:
This and the preceding paragraph are mostly the same. I think you meant to keep the first one?

Apart from that, I would reorder the middle part about CPU performance:

>  There are implementation tradeoffs to GPU-based dash processing, including potentially a larger number of dispatch stages, and we were able to achieve acceptable performance with the dash processing on CPU. Thus, whether moving dashing to a GPU stage is a performance win depends on the scene and other factors in the overall rendering pipeline. Even so, for scenes that have a very large number of dashed strokes, we would expect further performance improvement from GPU-based dashing.

"There are implementation tradeoffs to GPU-based dash processing, including potentially a larger number of dispatch stages. _Whether moving dashing to a GPU stage is a performance win depends on the scene and other factors in the overall rendering pipeline. While we would expect further performance improvement from GPU-based dashing, we were able to achieve acceptable performance with dash processing on CPU._